doc/dev/kdb-record.md

```diff
@@ -0,0 +1,110 @@
+# `kdb record` Command
+
+The `kdb record` tooling enables the user to record configuration sessions. It also allows exporting and importing the configuration changes.
+
+Recording sessions are stored on a per-user basis. This implies that a recording session can only track changes made by a single user.
+
+## Session handling
+
+* `kdb record start [session_name]`: Starts a new recording session. If no session name is provided, a new random name is generated. If a session with the provided name already exists, changes will be appended.
+* `kdb record stop`: Stops the current recording session. 
+* `kdb record delete [session_name]`: Deletes the specified recording session. If no session name is provided, the current session will be deleted. Deleting a session will not stop the recording. 
+* `kdb record list-sessions`: Lists all sessions by their name
+* `kdb record changes [session_name]`: Shows all changes for the specified session. If no session name is provided, the current session will be used. 
+
+## Exporting configuration changes
+
+`kdb record export [session_name] [--format <format>]`: Will dump the changed keys of the specified session onto `stdout`. If no session name is provided, the current session will be used. 
+
+As example, suppose the key `user:/my/app/hello` has the value `world` and you execute the following commands:
+* `kdb set user:/my/app/hello max`
+* `kdb set user:/my/app/hello susi`
+
+Only the value `susi` will be exported. Additionally, if you execute `kdb set user:/my/app/hello world` the key will not be exported at all, as there are no changes compared to the beginning of the recording.
+
+The same principle applies to the creation and deletion of keys within a session. If you create a key and delete it in the same session, it will not be exported.
+
+By default, the elektra dump format is used. The optional --format parameter can be used to specify which storage plugin should be used to format the output.
+
+## Importing configuration changes
+
+The output of the `kdb record export` command can be imported with the existing `kdb import` functionality. Depending on the specified storage plugin, other steps may be necessary, e.g. the Ansible plugins creates an Ansible playbook.
```

Comment:
I don' think this is as easy as you make it out to be. I'm pretty sure you need at least a special `kdb import --strategy`, if not a separate `kdb record import`.

See below (Tooling) for more.